index.js

```diff
@@ -16,13 +16,17 @@ module.exports = function retryAsPromised(callback, options) {
     timeout: options.timeout || undefined,
     match: options.match || [],
     backoffBase: options.backoffBase === undefined ? 100 : options.backoffBase,
-    backoffExponent: options.backoffExponent || 1.1
+    backoffExponent: options.backoffExponent || 1.1,
+    report: options.report || false,
+    reportFn: options.reportFn || console.log,
+    name:  callback.name || options.name || 'unknown'
   };
   
   // Massage match option into array so we can blindly treat it as such later
   if (!Array.isArray(options.match)) options.match = [options.match];
 
-  debug('Trying '+callback.name+' (%s)', options.$current);
+  debug('Trying '+ options.name+' (%s)', options.$current);
+  if(options.report) options.reportFn('Trying ' + options.name + ' #' + options.$current + ' at ' + new Date().toLocaleTimeString());
```

Comment:
To maintain bc lets scrap `reportFn` and just use `options.report` if defined.
And i think `options.report` should take two arguments, a log string and then meta data `{name, current, timestamp}`